Azure/open-service-broker-azure
# Fix Service Broker Update API

_This pull request was originally opened by @zhongyi-zhang_

**In this PR**:
* Fix Service Broker Update API
* Fix update method in CLI
(Above two should be a part of https://github.com/deis/azure-service-broker/pull/199)
<del>
* Add update features (refer to MASB)
  * Allow to update stored administratorLogin and password (Mainly for existing server scenario)
  * Allow to change service plan
* Add an optional provisioning parameter to enable TDE
(Above two fix https://github.com/deis/azure-service-broker/issues/220)


**Test**:

1. create a MSSQL instance with plan Basic

```
go run contrib/cmd/cli/*.go -H localhost -u username -P password provision -sid fb9bc99e-0aa9-11e6-8a8a-000d3a002ed5 -pid 3819fdfa-0aaa-11e6-86f4-000d3a002ed5 --param server=<redacted> --poll

Provisioning service instance 5ba763eb-851e-4fea-8455-4ca13a37df03

.........................

Service instance 5ba763eb-851e-4fea-8455-4ca13a37df03 has been successfully provisioned
```

2. update the instance with plan S0

```
go run contrib/cmd/cli/*.go -H localhost -u username -P password update -sid fb9bc99e-0aa9-11e6-8a8a-000d3a002ed5 -pid 2497b7f3-341b-4ac6-82fb-d4a48c005e19 -iid 5ba763eb-851e-4fea-8455-4ca13a37df03 --poll

Updating service instance 5ba763eb-851e-4fea-8455-4ca13a37df03

.............

Service instance 5ba763eb-851e-4fea-8455-4ca13a37df03 has been successfully updated
```
3. Check the pricing tier on Azure Portal

  The pricing tier is S0.
</del>
